benches/nexmark/generator/people.rs

```diff
@@ -0,0 +1,275 @@
+        state: next_us_state(rng),
+        date_time: DateTime::UNIX_EPOCH + std::time::Duration::from_millis(timestamp),
+        extra: String::new(),
+    }
+}
+
+/// Return a random person id (base 0).
+///
+/// Choose a random person from any of the 'active' people, plus a few 'leads'.
+/// By limiting to 'active' we ensure the density of bids or auctions per person
+/// does not decrease over time for long running jobs.  By choosing a person id
+/// ahead of the last valid person id we will make newPerson and newAuction
+/// events appear to have been swapped in time.
+///
+/// NOTE: The above is the original comment from the Java implementation. The
+/// "base 0" is referring to the fact that the returned Id is not including the
+/// FIRST_PERSON_ID offset, and should really be "offset 0".
+
+pub fn next_base0_person_id<R: Rng + ?Sized>(
+    conf: &config::Config,
+    event_id: Id,
+    rng: &mut R,
+) -> Id {
+    let num_people = last_base0_person_id(conf, event_id) + 1;
+    let active_people = std::cmp::min(num_people, config::NUM_ACTIVE_PEOPLE);
+    let n = rng.gen_range(0..(active_people + config::PERSON_ID_LEAD));
+    num_people - active_people + n
+}
+
+/// Return the last valid person id (ignoring FIRST_PERSON_ID). Will be the
+/// current person id if due to generate a person.
+pub fn last_base0_person_id(conf: &config::Config, event_id: Id) -> Id {
+    let epoch = event_id / conf.total_proportion();
+    let mut offset = event_id % conf.total_proportion();
+
+    if offset >= conf.person_proportion {
+        // About to generate an auction or bid.
+        // Go back to the last person generated in this epoch.
+        offset = conf.person_proportion - 1;
+    }
+    // About to generate a person.
+    epoch * conf.person_proportion + offset
+}
+
+// Return a random US state.
+fn next_us_state<R: Rng + ?Sized>(rng: &mut R) -> String {
+    US_STATES.choose(rng).unwrap().to_string()
+}
+
+// Return a random US city.
+fn next_us_city<R: Rng + ?Sized>(rng: &mut R) -> String {
+    US_CITIES.choose(rng).unwrap().to_string()
+}
+
+// Return a random person name.
+fn next_person_name<R: Rng + ?Sized>(rng: &mut R) -> String {
+    format!(
+        "{} {}",
+        FIRST_NAMES.choose(rng).unwrap(),
+        LAST_NAMES.choose(rng).unwrap()
+    )
+}
+
+// Return a random email address.
+fn next_email<R: Rng + ?Sized>(rng: &mut R) -> String {
+    format!("{}@{}.com", next_string(rng, 7), next_string(rng, 5))
+}
+
+// Return a random credit card number.
+fn next_credit_card<R: Rng + ?Sized>(rng: &mut R) -> String {
+    format!(
+        "{:04} {:04} {:04} {:04}",
+        rng.gen_range(0..10_000),
+        rng.gen_range(0..10_000),
+        rng.gen_range(0..10_000),
+        rng.gen_range(0..10_000)
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::config::Config;
+    use clap::Parser;
+    use rand::rngs::mock::StepRng;
+    use serial_test::serial;
+
+    #[test]
+    #[serial]
```

Comment:
Why do the tests need to be run in serial, there's no global state that I can see